test/src/kbasesearchengine/test/search/ElasticIndexingStorageTest.java

```diff
@@ -282,55 +295,134 @@ public void testGenome() throws Exception {
+    public void testMultiTypeSearchValidation1() throws Exception {
+
+        // null list of types
+        Set<GUID> guids = indexStorage.searchIds(null,
+                MatchFilter.create().withAccessGroupId(1),
+                null, AccessFilter.create().withAdmin(true));
+    }
+
+    @Test
+    public void testMultiTypeSearchValidation2() throws Exception {
+        indexObject("Genome", "genome01", new GUID("WS:1/1/1"), "MyGenome.1");
+
+        // empty list
+        Set<GUID> guids = indexStorage.searchIds(new ArrayList<String>(),
+                MatchFilter.create().withAccessGroupId(1),
+                null, AccessFilter.create().withAdmin(true));
+
+        Assert.assertTrue(guids.size() > 1);
+    }
+
+    @Test(expected = IOException.class)
+    public void testMultiTypeSearchValidation3() throws Exception {
+
+        // list containing a mix of null and non-null types
+        List<String> objectTypes = new ArrayList<>();
+        objectTypes.add(null);
+        objectTypes.add("Narrative");
+        Set<GUID> guids = indexStorage.searchIds(objectTypes,
+                MatchFilter.create().withAccessGroupId(1),
+                null, AccessFilter.create().withAdmin(true));
+    }
+
+    @Test
+    public void testMultiTypeSearchValidation4() throws Exception {
+        indexObject("Genome", "genome01", new GUID("WS:1/1/1"), "MyGenome.1");
+        List<String> objectTypes;
+        Set<GUID> guids;
+
+        // list with single non-null element
+        objectTypes = new ArrayList<>();
+        objectTypes.add("Genome");
+        guids = indexStorage.searchIds(objectTypes,
+                MatchFilter.create().withAccessGroupId(1),
+                null, AccessFilter.create().withAdmin(true));
+
+        Assert.assertEquals(1, guids.size());
+    }
+
+    @Test(expected = IOException.class)
+    public void testMultiTypeSearchValidation5() throws Exception {
+
+        // list exceeding max size
+        List<String> objectTypes = new ArrayList<>();
+        for(int ii=0; ii< ElasticIndexingStorage.MAX_OBJECT_TYPES_SIZE+1; ii++) {
+            objectTypes.add("Narrative");
+        }
+
+        Set<GUID> guids = indexStorage.searchIds(objectTypes,
+                MatchFilter.create().withAccessGroupId(1),
+                null, AccessFilter.create().withAdmin(true));
+    }
+
+    @Test
+    public void testMultiTypeSearch() throws Exception {
+        indexObject("Genome", "genome01", new GUID("WS:1/3/1"), "MyGenome.1");
+        indexObject("Assembly", "assembly01", new GUID("WS:1/4/1"), "MyAssembly.1");
+
+        // saerch for Genome and Assembly types
+        Set<GUID> guids = indexStorage.searchIds(ImmutableList.of("Genome", "Assembly"),
+                MatchFilter.create().withAccessGroupId(1),
+                  null, AccessFilter.create().withAdmin(true));
+
+        Assert.assertTrue(guids.contains(GUID.fromRef("WS", "1/3/1")));
+        Assert.assertTrue(guids.contains(GUID.fromRef("WS", "1/4/1")));
+    }
```

Comment:
I think a test to show that if a type is not in the type list objects of that type don't show up in the search is needed.

Reply:
Test has been strengthened.

Reply:
Unless I'm reading this test incorrectly it seems to be showing that a non-requested type *is* in the results which is not expected

Reply:
The side effect from the ObjectLookupProvider implementation in the prepare method is causing this. The test is correct and I have tried to clarify this doubt by breaking up the test into three stages - {Genome}, {Genome, Assembly}, {Genome, Assembly, AssemblyContig}. Keep in mind that Assemby and Assembly contig objects get indexed as a side-effect to indexing the Genome object. This test class needs to be cleaned up to remove this ugly side-effect.

Reply:
Clarification from OOB conversation: The original test implementation indexed a Genome, which also causes an Assembly and an AssemblyContig to be indexed (I assume this so the the Genome reference works - if the Assembly doesn't get indexed I presume the Genome will fail to index[?]). Then the test indexes *another* Assembly and AssemblyContig. Hence when the test searches for assemblies and genomes, it gets back 3 objects vs. the intuitive two. The original comment that one of the objects is an AssemblyContig is incorrect.

Reply:
Now the test just indexes a Genome (causing the A and AC to be indexed) and the results make more sense.